src/map.cpp

```diff
@@ -4377,30 +4377,43 @@ static void process_vehicle_items( vehicle &cur_veh, int part )
         }
     }
 
-    if( cur_veh.part_with_feature( part, VPFLAG_RECHARGE, true ) >= 0 &&
-        cur_veh.has_part( "RECHARGE", true ) ) {
+    if( cur_veh.part_with_feature( part, VPFLAG_RECHARGE, true ) >= 0 ) {
         for( auto &n : cur_veh.get_items( part ) ) {
             if( !n.has_flag( "RECHARGE" ) && !n.has_flag( "USE_UPS" ) ) {
                 continue;
             }
             // TODO: BATTERIES this should be rewritten when vehicle power and items both use energy quantities
             if( n.ammo_capacity() > n.ammo_remaining() ||
                 ( n.type->battery && n.type->battery->max_capacity > n.energy_remaining() ) ) {
-                // Around 85% efficient, so double discharge once every 7 seconds
-                const int per_charge = one_in( 7 ) ? 2 : 1;
-                const int missing = cur_veh.discharge_battery( per_charge, false );
-                if( missing < per_charge &&
-                    ( missing == 0 || x_in_y( per_charge - missing, per_charge ) ) ) {
-                    if( n.is_battery() ) {
-                        n.set_energy( 1_kJ );
-                    } else {
-                        n.ammo_set( "battery", n.ammo_remaining() + 1 );
+                if(cur_veh.part_flag( part, VPFLAG_ALTERNATOR_CHARGER ) ) {
+                    // Handcrank alternator charger
+                    const int power = -cur_veh.parts[part].info().power;
+                    const int chance = 48 * power / 100; // fairly low efficiency, like bicycle alternator
```

Comment:
No, this is not acceptable.  Hardcoding one solution's efficiency was bad, hardcoding two solutions' efficiency is just going to lead to a mess of stupid special cases.

Make this a JSON defined parameter, called 'efficiency' and recorded as floating point.  Read it from the JSON for both this and the normal recharger and then use the same logic for both.

Reply:
After digging around, I decided to use the `bonus` field in the JSON rather than introduce a new float.  The bonus value for the vehicle recharging station was unused, as far as I was able to determine.  I imagine at some point it was implying 600 watts of power to recharge things.